File: HackerRank-30-Days-Of-Code/Python/4.py
# Write a Person class with an instance variable, , and a constructor that takes an integer, , as a parameter. The constructor must assign  to  after confirming the argument passed as  is not negative; if a negative argument is passed as , the constructor should set  to  and print Age is not valid, setting age to 0.. In addition, you must write the following instance methods:

# yearPasses() should increase the  instance variable by .
# amIOld() should perform the following conditional actions:
# If , print You are young..
# If  and , print You are a teenager..
# Otherwise, print You are old..



class Person:
    def __init__(self,initialAge):
        # Add some more code to run some checks on initialAge
        if initialAge < 0:
            print("Age is not valid, setting age to 0.")
            self.age = 0
        else:
            self.age = initialAge

    def amIOld(self):
        # Do some computations in here and print out the correct statement to the console
        
        if self.age < 13:
            print("You are young.")
        elif  self.age >= 13 and self.age < 18:
            print("You are a teenager.")
        else:
            print("You are old.")


    def yearPasses(self):
        # Increment the age of the person in here
        self.age += 1

t = int(input())

File: HackerRank-30-Days-Of-Code/JavaScript/4.js
function Person(initialAge){
    if(initialAge<0){
        console.log("Age is not valid, setting age to 0.");
    }
    // Add some more code to run some checks on initialAge
  this.amIOld=function(){
   if(initialAge<13){
        console.log("You are young.");
    }else if(initialAge>=13 && initialAge<18){
        console.log("You are a teenager.");
    }else{
        console.log("You are old.");
    }
  };
   this.yearPasses=function(){
         initialAge+=1;
   };
}